src/libcollections/slice.rs

```diff
@@ -1087,7 +1086,7 @@ impl<T> [T] {
     /// elements.
     ///
     /// This sort is stable and `O(n log n)` worst-case but allocates
-    /// approximately `2 * n`, where `n` is the length of `self`.
+    /// approximately `n / 2`, where `n` is the length of `self`.
```

Comment:
I haven't gotten a chance to read all of the code yet, but I was wondering why big-O notation is used for the time complexity, but not for space? Is "approximately `n / 2`" not asymptotic behavior (e.g. is it an empirical number)? If not, then, I think it would be good to explicitly mention this.